lib/ansible/plugins/filter/vault.yml

```diff
@@ -32,15 +32,32 @@ DOCUMENTATION:
       default: False
 
 EXAMPLES: |
-  # simply encrypt my key in a vault
+  # Encrypt a value using the vault filter
   vars:
-    myvaultedkey: "{{ keyrawdata|vault(passphrase) }} "
+    myvaultedkey: "{{ 'my_secret_key' | vault('my_vault_password') }}"
 
-  - name: save templated vaulted data
-    template: src=dump_template_data.j2 dest=/some/key/vault.txt
-    vars:
-      mysalt: '{{2**256|random(seed=inventory_hostname)}}'
-      template_data: '{{ secretdata|vault(vaultsecret, salt=mysalt) }}'
+  # Encrypt a value and save it to a file using the template module
+  vars:
+    template_data: "{{ 'my_sensitive_data' | vault('another_vault_password', salt=(2**256 | random(seed=inventory_hostname))) }}"
+
+  - name: Save templated vaulted data
+    template:
+      src: dump_template_data.j2
+      dest: /some/key/vault.txt
+
+  # The dump_template_data.j2 file should contain:
+  # Encrypted secret: {{ template_data }}
+
+  # Encrypt a value and store it in a YAML file
+  - name: Write encrypted secret to a file
+    lineinfile:
+      create: yes
+      line: "encrypted_secret: {{ 'another_secret' | vault('vault_passphrase') }}"
+      dest: "/etc/ansible/vaulted_secrets.yml"
+
+  # Decrypt the vaulted data using unvault
```

Comment:
unvault has it's own examples, its already linked in the see also

Reply:
removed the unvault example since it’s already covered elsewhere